sql/commitment.sql

```diff
@@ -0,0 +1,667 @@
+		commitment_code char(4),
+		typ_c char(3),
+		typ_c_name text,
+		plannedcommit_ccnonexempt numeric,
+		plannedcommit_ccexempt numeric,
+		plannedcommit_citycost numeric,
+		plannedcommit_nccstate numeric,
+		plannedcommit_nccfederal numeric,
+		plannedcommit_nccother numeric,
+		plannedcommit_noncitycost numeric,
+		plannedcommit_total numeric
+	);
+	
+	CREATE TABLE project_source (
+		ccp_version text,
+		m_a_proj_id text,
+		m_agency_acro text,
+		m_agency char(3),
+		m_agency_name text,
+		description text,
+		proj_id text,
+		min_date date,
+		max_date date,
+		type_category text,
+		plannedcommit_ccnonexempt numeric,
+		plannedcommit_ccexempt numeric,
+		plannedcommit_citycost numeric,
+		plannedcommit_nccstate numeric,
+		plannedcommit_nccfederal numeric,
+		plannedcommit_nccother numeric,
+		plannedcommit_noncitycost numeric,
+		plannedcommit_total numeric,
+		adopt_ccnonexempt numeric,
+		adopt_ccexempt numeric,
+		adopt_citycost numeric,
+		adopt_nccstate numeric,
+		adopt_nccfederal numeric,
+		adopt_nccother numeric,
+		adopt_noncitycost numeric,
+		adopt_total numeric,
+		allocate_ccnonexempt numeric,
+		allocate_ccexempt numeric,
+		allocate_citycost numeric,
+		allocate_nccstate numeric,
+		allocate_nccfederal numeric,
+		allocate_nccother numeric,
+		allocate_noncitycost numeric,
+		allocate_total numeric,
+		commit_ccnonexempt numeric,
+		commit_ccexempt numeric,
+		commit_citycost numeric,
+		commit_nccstate numeric,
+		commit_nccfederal numeric,
+		commit_nccother numeric,
+		commit_noncitycost numeric,
+		commit_total numeric,
+		spent_ccnonexempt numeric,
+		spent_ccexempt numeric,
+		spent_citycost numeric,
+		spent_nccstate numeric,
+		spent_nccfederal numeric,
+		spent_nccother numeric,
+		spent_noncitycost numeric,
+		spent_total numeric,
+		spent_total_checkbooknyc numeric
+	);
+	
+COMMIT;
+-- end "commitment_source_tables"
+
+
+-- Break into "import_commitment" ETL sql file
+-- Populate source tables
+BEGIN;
+	COPY project_source FROM 'cpdb_projects_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+BEGIN;
+	COPY commitment_source FROM 'cpdb_planned_commitments_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+-- end "import_commitment"
+
+-- Remove "structure_target_tables" code
+-- The role of structuring the tables will be replaced by the target database
+-- Sharing its schema with this engine database
+-- Structure of target tables
+BEGIN;
+	CREATE TABLE managing_code (
+		id char(3) PRIMARY KEY
+	);
+
+	CREATE TABLE agency (
+		initials text PRIMARY KEY,
+		name text
+	);
+	
+	CREATE TYPE project_category AS ENUM (
+		'Fixed Asset',
+		'Lump Sum',
+		'ITT, Vehicles, and Equipment'
+	);
+		
+	CREATE TABLE project (
+		PRIMARY KEY (managing_code, id),
+    	managing_code char(3) references managing_code(id),
+		id text,
+		managing_agency text references agency(initials),
+		description text,
+		min_date date,
+		max_date date,
+		category project_category,
+		li_ft_m_pnt geometry(multipoint, 2263),
```

Comment:
What does the `m` in these column names represent? Meters?

Reply:
multi
This conversation is itself evidence that unfamiliar abbreviations are an anti-pattern, lol.

At a higher level, I've never liked this convention. It describes the shape of the data and not the context. It tells us what we already know from the schema (that it's a multi-point in long island state plane) and not why we should have this column (for calculations).

Maybe the columns should be something like "calculation" instead of "li_ft" and "tile" instead of "mercator". Anyway, this is probably a discussion for another time. For now, `m` is `multi`

Reply:
Yeah I'm not really a fan of encoding metadata about the column in the name because, as you pointed out, that's what the schema is for. It's basically [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), which is yucky.

Reply:
I guess one could argue that naming columns according to their projection like `li_ft` is doing the same thing, but I was OK with that because we needed a way to differentiate Mercator and LI geometry columns and looking at the schema only tells you the SRID which are hard to keep straight - the de facto standard `geom` column assumes you only have geometry column per table.

Reply:
If naming it after the projection is basically hungarian notation then naming it 'geom' is literally hungarian notation- `geom` being an abbreviation for the `geometry` type.

Reply:
Tbh I never though of it that way - I always thought of it as referring to just the "geometry" of the thing the database row represents. As in, the "geometry" of a tax lot is a property of the real thing just like the "lot" is. Now I'm curious where that convention comes from 😆 In any case, I'm not fighting you on the `li_ft` and `mercator` thing.

It sounds like naming conventions for geospatial columns is something we'll want to revisit but I'm comfortable putt that off if you are. For now, I say either stick with what you have or go with something like `_multipoint` to eliminate the ambiguity in the short term.